pkg/destroy/alibabacloud/alibabacloud.go

```diff
@@ -1256,21 +1281,48 @@ func (o *ClusterUninstaller) deleteDNSRecords() (err error) {
 		return
 	}
 
-	records, err := o.listRecord(baseDomain)
+	recordSetKey := func(recordType string, rr string) string {
+		return fmt.Sprintf("%s %s", recordType, rr)
+	}
+
+	// Get the parsing record of privatezone and delete the record in publiczone
+	// When the user manually deletes the private zone and records, it may cause the public records to be leaked.
+	privateZoneID, err := o.getPrivateZoneID()
+	if err != nil {
+		return
+	}
+	if privateZoneID == "" {
+		o.Logger.Info("The private zone ID is not obtained, and the DNS public records cannot be deleted")
+		return nil
+	}
+
+	privateRecords := map[string]bool{}
+	records, err := o.listPrivateZoneRecords(privateZoneID)
+	if err != nil {
+		return
+	}
+	for _, record := range records {
+		key := recordSetKey(record.Type, fmt.Sprintf("%s.%s", record.Rr, clusterName))
+		privateRecords[key] = true
+	}
+
+	publicRecords, err := o.listRecord(baseDomain)
 	if err != nil {
 		return
 	}
-	if len(records) == 0 {
+	if len(publicRecords) == 0 {
 		return
 	}
 
 	o.Logger.Debug("Start to delete DNS records")
-	for _, record := range records {
-		err = o.deleteRecord(record.RecordId)
-		if err != nil {
-			err = errors.Wrap(err, fmt.Sprintf("DNS record %q", record.RecordId))
-			o.Logger.Info(err)
-			return
+	for _, record := range publicRecords {
+		key := recordSetKey(record.Type, record.RR)
+		if privateRecords[key] {
+			err = o.deleteRecord(record.RecordId)
+			if err != nil {
+				err = errors.Wrap(err, fmt.Sprintf("DNS record %q", record.RecordId))
```

Comment:
When there is an error deleting the record, remove the key from `privateRecords` so that we don't wait the full minute for the record to be deleted. But also capture that there was an error so that `deleteDNSRecords` returns an error later after the wait.

Reply:
Have updated^^